Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+        return result
+
+    @classmethod
+    def from_file(cls, filename):
+        """Read a Table from a file.
+
+        Parameters
+        ----------
+        filename : str
+            The relative or absolute filename for the file from which to read the data.
+
+        Returns
+        -------
+        TableBase
+            A table constructed from the given file.
+        """
+        from Orange.data.io import FileFormat
+        from Orange.data import dataset_dirs
+
+        absolute_filename = FileFormat.locate(filename, dataset_dirs)
+        reader = FileFormat.get_reader(absolute_filename)
+        data = reader.read()
+
+        # Readers return plain table. Make sure to cast it to appropriate
+        # (subclass) type
+        if cls != data.__class__:
+            data = cls(data)
+
+        data.name = os.path.splitext(os.path.split(filename)[-1])[0]
+        data.__file__ = absolute_filename
+        return data
+
+    @classmethod
+    def from_url(cls, url):
+        """Read a table from a URL.
+
+        Parameters
+        ----------
+        url : str
+            The URL to read the table from.
+
+        Returns
+        -------
+        TableBase
+            A table constructed from the given URL.
+        """
+        from Orange.data.io import UrlReader
+        reader = UrlReader(url)
+        data = reader.read()
+        if cls != data.__class__:
+            data = cls(data)
+        return data
+
+    def _to_numpy(self, X=False, Y=False, meta=False, writable=False):
+        """Export a transformed numpy matrix.
+
+        The order is always X, Y, meta. Always 2D.
+        The columns are in the same order as in Table.domain._.
+        If writable == False (default), the numpy writable flag is set to false.
+        This means write operations on this array will loudly fail.
+
+        This is not the same as .values, but instead transforms the attributes to
+        numeric values (where possible); e.g. uses var.values indices instead of
+        actual descriptors.
+
+        Parameters
+        ----------
+        X : bool, default False
+            Whether to include the domain attributes in the result.
+        Y : bool, default False
+            Whether to include the domain class variables in the result.
+        meta : bool, default False
+            Whether to include the domain metas in the result.
+        writable : bool, default False
+            Whether to mark the resulting domain as writable.
+        Returns
+        -------
+        np.ndarray
+            The numpy array of the selected and transformed table data.
+        """
+        cols = []
+        cols += self.domain.attributes if X else []
+        cols += self.domain.class_vars if Y else []
+        cols += self.domain.metas if meta else []
+
+        # support using this in TableSeries, whose len gives the number of columns
+        n_rows = 1 if isinstance(self, SeriesBase) else len(self)
```

Comment:
`len(table[col])` gives what? Length of `pd.Series` object equals the number of rows.